contracts/Vault2.sol

```diff
@@ -1,6 +1,26 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
-contract Vault2 {
-    
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract Vault2 is ERC20 {
+    event Minted(address owner, uint amount);
+    event Burned(address owner, uint amount);
+
+    constructor() 
+        ERC20("Vault2", "VALT2") {}
+
+    function mint(uint _amount) public payable {
+        require(_amount != 0, "Must mint a non-zero value.");
+        require(msg.value == _amount, "Must send exactly the same amount of ether.");
+        _mint(msg.sender, _amount);
+        emit Minted(msg.sender, _amount);
```

Comment:
_mint already emits an event indicating a `mint` - https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L267. `Minted` event can be removed